### PR TITLE
Invite and Reject Flows

### LINKED ISSUE #415: Invitation Flow Error
I am `bryan@rerum.io`.  I invite `thehabes1110@gmail.com` to join a project, `thehabes1110@rerum.io` is not a RERUM User yet.

I am `thehabes1110@gmail.com`.  I send all E-mails from `bryan@rerum.io` straight to junk because I don't like them.

I am `thehabes1110@gmail.com` and I never saw that Email.  I go to TPEN3 and decide to make an account.  I am added to the RERUM Auth0 Tenant correctly and an agent is generated for me.  I click Accept and am routed back to TPEN3.  I experience an error because the app tries to add a user with duplicate Emails.  There is nothing I can do, and I cannot use TPEN3 at all.

### PR DESCRIPTION
**Pull request by thehabes**
## Fix: Handle invited users who sign up independently

  Resolves #415

  ### Problem

  Users invited to TPEN3 projects who create accounts independently (before accepting their invitation) encounter a "duplicate email" error that locks them out of the system. This occurs
  because:

  1. User A invites `user@example.com` to a project
  2. A temporary user is created with `user@example.com`
  3. The invitee ignores the email and independently creates a TPEN3 account with `user@example.com`
  4. On signup, the system tries to create a user with an email that already exists
  5. Duplicate email error prevents account creation → user is locked out

  ### Solution

  This PR implements a **temporary user merge system** that automatically reconciles invited users with their actual accounts:

  1. **Temporary users** are created with an `inviteCode` field when invited
  2. **On signup**, if a temp user exists with the same email:
     - The real user account is created
     - All group memberships are transferred from temp user to real user
     - The temporary user is deleted
  3. **Multi-project invites** are handled by reusing temp users across projects
  4. **Smart deletion** only removes temp users when they have no remaining group memberships

  ### Changes

  #### Core Files Modified

  **`auth/index.js`** - Authentication middleware
  - Detect temporary users by email when new users authenticate
  - Automatically merge temp user memberships into new user accounts
  - Handle email conflicts with appropriate 409 error response
  - Clean up `inviteCode` field on authenticated users

  **`classes/User/User.js`** - User model
  - Added `mergeFromTemporaryUser()` method to transfer group memberships
  - Handles deletion of temp users after merge

  **`classes/Group/Group.js`** - Group model
  - Added `transferMembership()` method to move memberships between users
  - Added `getGroupsByMember()` static method to find all groups for a user
  - Properly merges roles when target user already exists in group

  **`classes/Project/Project.js`** - Project model
  - Updated `inviteNewTPENUser()` to reuse existing temp users for multi-project invites
  - Updated `removeMember()` with smart temp user deletion (only deletes if no remaining memberships)

  ### How It Works

  #### Scenario 1: Single Project Invitation
  1. User invited to Project A → Create temp user (has inviteCode)
  2. User signs up independently → Detect temp user by email
  3. Create real user → Transfer Project A membership → Delete temp user
  4. ✅ User has access to Project A

  #### Scenario 2: Multi-Project Invitation
  1. User invited to Project A → Create temp user
  2. User invited to Project B → Reuse temp user, add to Project B
  3. User signs up independently → Detect temp user
  4. Transfer both memberships → Delete temp user
  5. ✅ User has access to both projects

  #### Scenario 3: Decline One of Multiple Invites
  1. User invited to Projects A and B (same temp user)
  2. User declines Project A → Remove from Project A group
  3. Check remaining groups → Still in Project B
  4. ✅ Temp user preserved (not orphaned)
  5. User accepts Project B → Merge and delete temp user

  ### Testing

  - ✅ All endpoint resiliency tests passed (10/10)
  - ✅ Service handles invalid IDs, malformed requests, and edge cases correctly
  - ✅ Authentication flow working properly
  - ✅ No breaking changes to existing functionality
  - ✅ Manually tested invite flow scenarios

  ### Security & Data Integrity

  - Email conflicts detected and handled with 409 error
  - Only temporary users (with `inviteCode`) can be auto-deleted
  - Real users are never auto-merged or deleted
  - Group membership transfers are atomic
  - Proper validation prevents accidental data loss

### CODE REVIEW

File: classes/Project/Project.js [hunk 1]
```diff
@@ -161,9 +165,27 @@ export default class Project {
...
+      tempUser = await existingUserLookup.getByEmail(email)
+    } catch (err) {
+      // No user found - that's fine, we'll create one
+    }
+
+    if (tempUser && tempUser.inviteCode) {
+      // Reuse existing temp user - just add to this project's group
+      await this.inviteExistingTPENUser(tempUser._id, roles)
+      return { "tpenUserID": tempUser._id, "tpenGroupID": this.data.group }
+    }
```
[the remark this answers is not in the document]
Comment (thehabes) [reply classes/Project/Project.js]: the concern identifies a theoretical vulnerability that requires direct database access, which would already constitute a critical security breach

**Comment by thehabes** [comment]
@claude review this PR
